File: backend/TravelBuddyAPI/Models/Place.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;

namespace TravelBuddyAPI.Models;

public abstract class Place
{
    [Required]
    [Key]
    public Guid Id { get; set; }

    [Required]
    public string? Name { get; set; }

    public string? Country { get; set; }

    public string? State { get; set; }

    public string? City { get; set; }

    public string? Street { get; set; }

    public string? HouseNumber { get; set; }

    [Precision(18, 8)]
    public decimal? Latitude { get; set; }

    [Precision(18, 8)]
    public decimal? Longitude { get; set; }

    public List<TripPointReview>? Reviews { get; set; }
}

File: app/types/Place.ts
import { Category, Condition } from "./Profile";

export interface PlaceCompact {
  id: string;
  providerId?: string;
  name: string;
  country: string;
  state?: string;
  city: string;
}

export interface PlaceViewModel {
  providerId: string;
  id: string;
  title?: string;
  subtitle?: string;
  imageUrl?: string;
}

export interface Place extends PlaceCompact {
  superCategoryId?: string;
  street?: string;
  houseNumber?: string;
  latitude?: number;
  longitude?: number;
}

export interface PlaceOverview extends PlaceCompact {
  street?: string;
  houseNumber?: string;
  superCategory: Category;
}

export interface PlaceDetails extends PlaceOverview {
  latitude: number;
  longitude: number;
  averageCostPerPerson: number;
  averageTimeSpent: string;
  averageRating: number;
  conditions: Condition[];
  categories: Category[];
}
